samples/server/go-server/api/server/storage/memstore.go

```diff
@@ -42,53 +43,62 @@ func NewMemStore() server.Storager {
 		occurrencesByID: map[string]*pb.Occurrence{},
 		notesByID:       map[string]*pb.Note{},
 		opsByID:         map[string]*opspb.Operation{},
-		projects:        map[string]bool{},
+		projects:        []*pb.Project{},
 	}
 }
 
 // CreateProject adds the specified project to the mem store
 func (m *memStore) CreateProject(pID string) error {
 	m.Lock()
 	defer m.Unlock()
-	if _, ok := m.projects[pID]; ok {
+	project := &pb.Project{Name: name.FormatProject(pID)}
+	if containsProject(m.projects, project) {
 		return status.Error(codes.AlreadyExists, fmt.Sprintf("Project with name %q already exists", pID))
 	}
-	m.projects[pID] = true
+	m.projects = append(m.projects, project)
 	return nil
 }
 
 // DeleteProject deletes the project with the given pID from the mem store
 func (m *memStore) DeleteProject(pID string) error {
 	m.Lock()
 	defer m.Unlock()
-	if _, ok := m.projects[pID]; !ok {
-		return status.Error(codes.NotFound, fmt.Sprintf("Project with name %q does not Exist", pID))
+	project := &pb.Project{Name: name.FormatProject(pID)}
+	index, err := findProject(m.projects, project)
+	if err != nil {
+		return status.Error(codes.NotFound, fmt.Sprintf("Project with name %q does not exist", pID))
 	}
-	delete(m.projects, pID)
+	copy(m.projects[index:], m.projects[index+1:])
+	m.projects[len(m.projects)-1] = nil
+	m.projects = m.projects[:len(m.projects)-1]
 	return nil
 }
 
 // GetProject returns the project with the given pID from the mem store
 func (m *memStore) GetProject(pID string) (*pb.Project, error) {
```

Comment:
Perhaps we should keep the map for efficient gets and then just use the sorted keys of the map for the List method? What do you think? 

This isn't a huge deal since its in memory and the size of things shouldn't be too large.

Reply:
Agree. Switching to lists turned out to be more intrusive than I first thought so going back to a map is better for several reasons.

Reply:
OMG. So much ugly code was introduced by this change. What was I thinking?!? Good thing you reviewed this.😄 

363992e reverts back to storing projects in a map.